packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -373,39 +395,63 @@ contract Deploy is Deployer {
+        DeployImplementations di = new DeployImplementations();
+        (DeployImplementationsInput dii, DeployImplementationsOutput dio) = di.etchIOContracts();
+
+        dii.set(dii.withdrawalDelaySeconds.selector, cfg.faultGameWithdrawalDelay());
+        dii.set(dii.minProposalSizeBytes.selector, cfg.preimageOracleMinProposalSize());
+        dii.set(dii.challengePeriodSeconds.selector, cfg.preimageOracleChallengePeriod());
+        dii.set(dii.proofMaturityDelaySeconds.selector, cfg.proofMaturityDelaySeconds());
+        dii.set(dii.disputeGameFinalityDelaySeconds.selector, cfg.disputeGameFinalityDelaySeconds());
+        string memory release = "dev";
+        dii.set(dii.release.selector, release);
+        dii.set(
+            dii.standardVersionsToml.selector, string.concat(vm.projectRoot(), "/test/fixtures/standard-versions.toml")
+        );
+        dii.set(dii.superchainConfigProxy.selector, mustGetAddress("SuperchainConfigProxy"));
+        dii.set(dii.protocolVersionsProxy.selector, mustGetAddress("ProtocolVersionsProxy"));
+        dii.set(dii.opcmProxyOwner.selector, cfg.finalSystemOwner());
 
-        // Fault proofs
-        deployOptimismPortal2();
-        deployDisputeGameFactory();
-        deployDelayedWETH();
-        deployPreimageOracle();
-        deployMips();
-    }
+        if (_isInterop) {
+            di = DeployImplementations(new DeployImplementationsInterop());
+        }
+        di.run(dii, dio);
 
-    /// @notice Deploy all of the implementations
-    function deployImplementationsInterop() public {
-        // TODO: Replace the actions in this function with a call to DeployImplementationsInterop.run()
-        console.log("Deploying implementations");
-        deployL1CrossDomainMessenger();
-        deployOptimismMintableERC20Factory();
-        deploySystemConfigInterop();
-        deployL1StandardBridge();
-        deployL1ERC721Bridge();
+        save("L1CrossDomainMessenger", address(dio.l1CrossDomainMessengerImpl()));
+        save("OptimismMintableERC20Factory", address(dio.optimismMintableERC20FactoryImpl()));
+        save("SystemConfig", address(dio.systemConfigImpl()));
+        save("L1StandardBridge", address(dio.l1StandardBridgeImpl()));
+        save("L1ERC721Bridge", address(dio.l1ERC721BridgeImpl()));
 
         // Fault proofs
-        deployOptimismPortalInterop();
-        deployDisputeGameFactory();
-        deployDelayedWETH();
-        deployPreimageOracle();
-        deployMips();
+        save("OptimismPortal2", address(dio.optimismPortalImpl()));
+        save("DisputeGameFactory", address(dio.disputeGameFactoryImpl()));
+        save("DelayedWETH", address(dio.delayedWETHImpl()));
+        save("PreimageOracle", address(dio.preimageOracleSingleton()));
+        save("Mips", address(dio.mipsSingleton()));
+
+        Types.ContractSet memory contracts = _impls();
+        ChainAssertions.checkL1CrossDomainMessenger({ _contracts: contracts, _vm: vm, _isProxy: false });
```

Comment:
I don't see `ChainAssertions.check*` calls in this section for `L1StandardBridge`, `L1ERC721Bridge`, and `MIPS`

Reply:
will add those

Reply:
91cb035b77ec